src/collective/elasticsearch/tests/test_search.py

```diff
@@ -69,180 +59,283 @@ def test_date_index_query(self):
 
-    def test_brains_indexing(self):
+
+class TestBrainsIndexing(BaseFunctionalTest):
+    def setUp(self):
+        super().setUp()
         for idx in range(120):
-            createObject(self.portal, "Document", f"{idx:04d}page", title=f"Page {idx}")
+            api.content.create(
+                self.portal, "Document", f"{idx:04d}page", title=f"Page {idx}"
+            )
+        self.commit(wait=1)
+        self.el_results = self.search(
+            {
+                "portal_type": "Document",
+                "sort_on": "getId",
+                "sort_order": "asc",
+            }
+        )
+
+    def test_all_indexed(self):
+        self.assertEqual(len(self.el_results), 120)
+
+    @parameterized.expand(
+        [
+            (0, "0000page"),
+            (-1, "0119page"),
+            (-50, "0070page"),
+            (-55, "0065page"),
+            (-100, "0020page"),
+        ]
+    )
+    def test_ordering(self, result_idx, expected):
+        self.assertEqual(self.el_results[result_idx].getId, expected)
+
+
+class TestCatalogRecordDeleted(BaseFunctionalTest):
+    def setUp(self):
+        super().setUp()
+        catalog = self.catalog._catalog
+        self.event = api.content.create(
+            self.portal, "Event", "event-test", title="Gone Event"
+        )
+        self.commit(wait=1)
+        path = "/".join(self.event.getPhysicalPath())
+        catalog.uncatalogObject(path)
         self.commit()
-        el_results = self.catalog(
-            portal_type="Document", sort_on="getId", sort_order="asc"
+
+    def test_search_results(self):
+        el_results = self.search({"portal_type": "Event", "Title": "Gone Event"})
+        self.assertEqual(len(el_results), 1)
+        brain = el_results[0]
+        self.assertTrue(ICatalogBrain.providedBy(brain))
+        self.assertEqual(brain.getRID(), -1)
+        # Test data from elastic will populate the brain
+        self.assertEqual(brain.portal_type, "Event")
+        self.assertEqual(brain.Title, "Gone Event")
+        # Test
+        self.assertEqual(brain.getPath(), "/plone/event-test")
+        self.assertEqual(brain.getURL(), "http://nohost/plone/event-test")
+        self.assertEqual(brain.getObject(), self.event)
+
+
+class TestDeleteObjectNotReflectedOnES(BaseFunctionalTest):
+    def setUp(self):
+        super().setUp()
+        catalog = self.catalog._catalog
+        self.event = api.content.create(
+            self.portal, "Event", "event-test", title="Gone Event"
         )
-        self.assertEqual(len(el_results), 120)
-        self.assertEqual(el_results[0].getId, "0000page")
-        self.assertEqual(el_results[-1].getId, "0119page")
-        self.assertEqual(el_results[-50].getId, "0070page")
-        self.assertEqual(el_results[-55].getId, "0065page")
-        self.assertEqual(el_results[-100].getId, "0020page")
+        self.commit(wait=1)
+        path = "/".join(self.event.getPhysicalPath())
+        catalog.uncatalogObject(path)
```

Comment:
Isn't it risky to keep the registry in Elastic, when Plone tells you to uncatalog an object? I believe that when someone does this, it is for the content not to appear in searches. What's the idea here?

Reply:
This test is to reproduce a situation we had with a client.
For some reason, there were cases where the object was not on Catalog anymore but still existed on ElasticSearch.
There is also the case when the object does not exist on the database anymore but still exists on ElasticSearch.

Reply:
But expected is uncatalogObject remove de record of Elastic. I think you have to use another method, created just for the test, which removes the Elastic record, to simulate this situation.

Reply:
You can even do a test in which `catalog.uncatalogObject(path)` is called for a record, and verify that it no longer appears in the search.

Reply:
The use case we have is: An object in the database, for some reason, is not in portal_catalog anymore **but** still exists on ElasticSearch. The ZCatalog.uncatalogObject is the closest method we have to reproducing this (as it is a low-level call that removes the object from the catalog but not from the ZODB or ElasticSearch).

Reply:
I think you can do the following:

- Patch `uncatalogObject` to remove the record from Elastic.
- Save original `uncatalogObject`.
- Use the original `uncatalogObject` in the test to simulate your situation.